### ISSUE TITLE
[Feature] Datetime selection dialog is confusing to users

### ISSUE THREAD

**Issue by Nek-12** [maxkeppeler/sheets-compose-dialogs#53]
We are not satisfied with the UX/UI of the datetime dialog. While we are using a custom implementation, you may want to address the following issue:

The dialog has no indication as to what the user must do to confirm that they've entered the value. This means that even if the user has scrolled the day / month / year values to their desired values, they can't tap "OK" until they have tapped each value (it starts being filled with a primary color). If the user taps "Select time" without confirming their date values, they will all be reset.
It took me as a developer 3 minutes to understand what's going on, imagine how long it will take a regular user to understand.

**Comment by maxkeppeler** [comment]
I like receiving feedback, but I believe that constructive criticism with suggestions for improvement would be more beneficial. I had two individuals with no prior experience in IT try out this component, and they were able to successfully select a date and time, without any hints or assistance and at first try. 

I don't think auto-selecting the current centered item after a specific duration would be a solution.
Feel free to provide me with any suggestions to improve this component.

**Comment by Nek-12** [comment]
Yes, after giving it some thought I wanted to express a possible solution. Currently have an idea for a pr in mind. 
When the dialog is scrolled, we can remove the confirmation at all by introducing snapping to the values. 
The dialog in this case would simply use the current value in the list. That's how most drum selectors work in apps nowadays. A good example is the Xiaomi's alarm app. 
A less Intrusive solution would be to consider the values currently on the drum "selected" when the user taps "select time" in the datetime dialog with time enabled, or at least save and restore them instead of dropping the values altogether. 
If we go with the first one, we'll also need to increase the height of the dialog a bit to allow for easier flinging.

**Comment by starry-shivam** [comment]
So, I looked into this and couldn't really figure out what's going wrong from my end to cause incorrect behavior in the Chinese locale. I browsed some open issues on `sheets-compose-dialogs` (the library I use to show calendar and datetime dialogs) and found this: https://github.com/maxkeppeler/sheets-compose-dialogs/issues/57. I think this may be the reason why it's not working properly in the Chinese locale.

Regarding the other issue of the incorrect AM/PM, I was indeed able to reproduce it on my real device (Xiaomi POCO F5/Redmi Note 12 Turbo), where it didn't work in both 24-hour format and 12-hour format (It didn't even show me the option to select AM/PM in 12-hour format). However, for some reason, it seems to work fine on Android Studio's emulator?! I really have no idea why. I think it's also some problem or bug with the library, though I can't say for sure.

I also found this issue: https://github.com/maxkeppeler/sheets-compose-dialogs/issues/53, which I personally agree with. It took me quite a few tries to understand how it's supposed to work, and I believe it could be confusing for users as well.

Considering all of this, I think it would be wise to switch to the standard Android library at least for the datetime picker, or maybe look at some other alternatives because there seem to be multiple issues with the current implementation.
